core/src/test/kotlin/ch/ergon/dope/buildTest/SubQueryTest.kt

```diff
@@ -21,9 +33,75 @@ class SubQueryTest {
 
         val actual: String = create
             .select(someStringField())
-            .from(create.selectAsterisk().from(someBucket()).alias("asdf"))
+            .from(create.selectAsterisk().from(someBucket()).queryAlias("asdf"))
             .build().queryString
 
         assertEquals(expected, actual)
     }
+
```

Comment:
can you add test with subqueries for at least one conditional-expression (decode or nvl2), typefunction (toNumber...), relational-expression (between) and useKeys

Reply:
I added mainly extension function tests and a few build tests. Let me know if you think I should add more.
Between is currently not possible, because ArrayTypes are not yet able to be compared.